luarules/gadgets/cmd_build_bugger_off.lua

```diff
@@ -0,0 +1,186 @@
+        if Spring.AreTeamsAllied(unitTeam, builderTeam) == false then
+            return false
+        end
+        if shouldNotBuggeroff[Spring.GetUnitDefID(interferingUnitID)] then
+            return false
+        end
+        if WillBeNearTarget(interferingUnitID, x, y, z, BUILDER_DELAY_SECONDS, radius) then
+            return true
+        end
+        if isInTargetArea(interferingUnitID, x, y, z, radius) then
+            return true
+        end
+        return false
+    end
+
+    local function distance(x1, z1, x2, z2)
+        local dx, dz = x2 - x1, z2 - z1
+        return math.diag(dx, dz)
+    end
+
+    function gadget:GameFrame(frame)
+        if frame % 10 ~= 0 then
+            return
+        end
+        for builderID, _ in pairs(watchedBuilders) do
+            local cmdID, options, tag, targetX, targetY, targetZ =  Spring.GetUnitCurrentCommand(builderID, 1)
+            local isBuilding  = false
+            local builderTeam = Spring.GetUnitTeam(builderID)
+            local x, y, z     = Spring.GetUnitPosition(builderID)
+            local targetID    = Spring.GetUnitIsBuilding(builderID)
+            if targetID then
+                isBuilding = true
+            end
+            if cmdID == nil or cmdID > -1 or isBuilding then
+                print("Clearing watched builder fast")
+                watchedBuilders[builderID]      = nil
+                builderRadiusOffsets[builderID] = nil
+                return
+            end
+            local builtUnitDefID = cmdID * -1
+            if distance(targetX, targetZ, x, z) > FAST_UPDATE_RADIUS then
+                print("Too far, demoting to slow")
+                watchedBuilders[builderID]    = nil
+                slowUpdateBuilders[builderID] = true -- Do distance checks less frequently
+            elseif distance(targetX, targetZ, x, z) > BUILDER_BUILD_RADIUS + cachedUnitDefs[builtUnitDefID].radius then
+                -- Check distance frequently once you're closer
+            else
+                -- Get list of units to check
+                local buggerOffRadius  = cachedUnitDefs[builtUnitDefID].radius + builderRadiusOffsets[builderID]
+                local searchRadius     = cachedUnitDefs[builtUnitDefID].radius + SEARCH_RADIUS_OFFSET
+                local interferingUnits = Spring.GetUnitsInCylinder(targetX, targetZ, searchRadius)
+
+                -- Escalate the radius every update. We want to send units away the minimum distance, but  
+                -- if there are many units in the way, they may cause a traffic jam and need to clear more room.
+                builderRadiusOffsets[builderID] = builderRadiusOffsets[builderID] + 5 
+
+                for i = 1, #interferingUnits do
+                    local interferingUnitID = interferingUnits[i]
+                    if builderID ~= interferingUnitID then
+                        local unitPosition = {Spring.GetUnitPosition(interferingUnitID)}
```

Comment:
```suggestion
                        local unitPositionX, _, unitPositionZ = Spring.GetUnitPosition(interferingUnitID)
```
Can save the table creation call, and remove the need to use cryptic numeric indexes when referencing these values.